src/task/SubscribeAndReceiveSubscribeAckTask.ts

```diff
@@ -88,14 +106,25 @@ export default class SubscribeAndReceiveSubscribeAckTask extends BaseTask {
       isSendingStreams,
       this.context.videoStreamIndex.localStreamDescriptions(),
       // TODO: handle check-in mode, or remove this param
-      true
+      true,
+      compressedSDPOffer
     );
     this.context.logger.info(`sending subscribe: ${JSON.stringify(subscribe)}`);
     this.context.signalingClient.subscribe(subscribe);
 
     const subscribeAckFrame = await this.receiveSubscribeAck();
     this.context.logger.info(`got subscribe ack: ${JSON.stringify(subscribeAckFrame)}`);
-    this.context.sdpAnswer = subscribeAckFrame.sdpAnswer;
+    let decompressedText: string = '';
+    if (subscribeAckFrame.compressedSdpAnswer.length > 0) {
+      decompressedText = this.textCompressor.Decompress(
+        subscribeAckFrame.compressedSdpAnswer,
+        this.context.previousSdpAnswerAsString
+      );
+    }
+    this.context.sdpAnswer =
```

Comment:
Should we throw an error and cancel the task here if decompress fail?

Reply:
Let me check what we do at other places.